src/wallet/rpc/spend.cpp

```diff
@@ -1406,6 +1406,11 @@ RPCHelpMan sendall()
                 }
             }
 
+            // Limit size
+            if (tx_size.weight > MAX_STANDARD_TX_WEIGHT) {
+                throw JSONRPCError(RPC_WALLET_INSUFFICIENT_FUNDS, "Transaction too large.");
+            }
```

Comment:
Might this check make more sense if it were placed in `FinishTransaction` (which is called by `send` and `sendall`) within the section that is only performed if the transaction is complete? The advantages of this would be that A) this check is now added for both `sendall` and `send`, and B) this check isn't performed if the transaction doesn't contain a complete set of signatures.

Reply:
[`send` is using `FundTransaction`](https://github.com/bitcoin/bitcoin/blob/19585eeb778ac093d7b9e93abd099629849c3fef/src/wallet/rpc/spend.cpp#L1228) which then calls the [`wallet::FundTransaction`](https://github.com/bitcoin/bitcoin/blob/19585eeb778ac093d7b9e93abd099629849c3fef/src/wallet/spend.cpp#L1087) which uses `CreateTransaction` therefore performing those checks.

Reply:
Ok, I hadn't noticed that.